server/src/test/java/io/crate/analyze/UpdateAnalyzerTest.java

```diff
@@ -639,14 +639,14 @@ public void test_update_array_of_objects_subarray_by_elements() throws IOExcepti
             .build();
 
         assertThatThrownBy(() -> e.analyze("update t set a['b'][1][1] = 10;"))
-            .isExactlyInstanceOf(UnsupportedOperationException.class)
-            .hasMessage("Nested array access is not supported");
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("cannot use expression \"a\"['b'][1][1] as a left side of an assignment");
         assertThatThrownBy(() -> e.analyze("update t set a[1][1]['b'] = 10;"))
-            .isExactlyInstanceOf(UnsupportedOperationException.class)
-            .hasMessage("Nested array access is not supported");
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("cannot use expression \"a\"[1][1]['b'] as a left side of an assignment");
         assertThatThrownBy(() -> e.analyze("update t set a[1]['b'][1] = 10;"))
-            .isExactlyInstanceOf(UnsupportedOperationException.class)
-            .hasMessage("Nested array access is not supported");
+            .isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("cannot use expression \"a\"[1]['b'][1] as a left side of an assignment");
```

Comment:
Just to clarify, are we handling these as a follow up?

Reply:
No, we don't allow updates via subscripts at all - if you look at the next few lines you'll see that we have tests forbidding this for plain string keys as well.  This change just brings the error messages in line with the other cases.